src/mongo/db/curop.h

```diff
@@ -305,6 +306,10 @@ class CurOp {
         if (_dbprofile <= 0)
             return false;
 
+        if (serverGlobalParams.rateLimit > 1 && _dbprofile >= 2 && ms < serverGlobalParams.slowMS) {
```

Comment:
Please declare another method and implement it in the cpp file so you can remove
`static PseudoRandom _prng` from members.